IBAnimatable/AnimatorFactory.swift

```diff
@@ -12,11 +12,11 @@ public struct AnimatorFactory {
                                   transitionDuration: Duration = defaultTransitionDuration) -> AnimatedTransitioning? {
     switch transitionAnimationType {
     case .systemRotate:
-      return SystemRotateAnimator(transitionDuration: transitionDuration)
+      return SystemTransitionAnimator(systemType: .rotate, duration: transitionDuration)
```

Comment:
I'm wondering if we can't simplify this switch for the system animation, maybe something like:

```swift
if let systemType = transitionAnimationType.systemType {
   SystemTransitionAnimator(systemType: transitionAnimationType, duration: transitionDuration)
   return // or making an else and splitting in two methods
}

switch transitionAnimationType {
case .portal:
   ...
case ...:
   ...
default:
   fatalError("No animator available for this transition.")
}
```

where ` transitionAnimationType.systemType` would be a simple mapping.

A second solution could be to just detect if the `transitionAnimationType` is a system animation or not, then let `SystemTransitionAnimator` makes the mapping job.